### DISCUSSION TITLE
Proper way to 'unload' a loaded wasm module

### DISCUSSION THREAD

**Discussion by sampaioletti** [wasmi-labs/wasmi#1301] [Q&A] [answered]
What would be the proper/optimum way to 'unload' or 'drop' a module with the intent to load a different module. 

e.g.

1. Keep `Engine` but drop `Module` `Store` and `Linker`
2. Drop all
3. ... Some other combination

Since 'almost' everything keeps a hierarchy of references I'm unsure what would be most efficient i.e. freeing the right resources without unnecessary re-initialization.

**Comment by Robbepop** [comment, marked as answer]
You can load multiple modules using the same Wasmi `Engine`. Wasmi `Engine` mostly holds information about a module's function code. All other information is stored in the `Store`. You are right that it is currently not possible to "unload" a Wasm module from an `Engine`. This has technical reasons. If this use case is common enough we might be able to add this functionality.

So if it is important to unload a Wasmi `Module` you'd need to drop `Module`, `Store` and `Linker`.

**Reply by sampaioletti** [reply]
Thank you for the response. That clears it up.

Thanks for all the work on the library, it works great for us.

**Reply by Robbepop** [reply]
Happy to hear that! :)
